src/test/regress/expected/alter_role_propagation.out

```diff
@@ -237,12 +237,141 @@ SELECT run_command_on_workers($$SELECT row(rolname, rolsuper, rolinherit,  rolcr
  (localhost,57638,t,"(alter_role_1,f,f,f,f,f,f,f,0,md5be308f25c7b1a2d50c85cf7e6f074df9,2052)")
 (2 rows)
 
--- table belongs to a role
--- we don't support propagation of configuration_parameters and notice the users
-ALTER ROLE alter_role_1 SET enable_hashagg TO FALSE;
-NOTICE:  Citus partially supports ALTER ROLE for distributed databases
-DETAIL:  Citus does not propagate ALTER ROLE ... SET/RESET commands to workers
-HINT:  You can manually alter roles on workers.
+-- give login permissions so that we can connect and check if the previous queries were propagated
+ALTER ROLE alter_role_1 LOGIN CONNECTION LIMIT 10;
+-- alter configuration_parameter defaults for a user
+ALTER ROLE CURRENT_USER SET enable_hashagg TO FALSE;
+SELECT run_command_on_workers('SHOW enable_hashagg');
+ run_command_on_workers
+---------------------------------------------------------------------
+ (localhost,57637,t,off)
+ (localhost,57638,t,off)
+(2 rows)
+
+-- reset to default values
+ALTER ROLE CURRENT_USER RESET enable_hashagg;
+SELECT run_command_on_workers('SHOW enable_hashagg');
+ run_command_on_workers
+---------------------------------------------------------------------
+ (localhost,57637,t,on)
+ (localhost,57638,t,on)
+(2 rows)
+
+-- provide role and database names
+ALTER ROLE alter_role_1 IN DATABASE regression SET enable_hashjoin TO 0;
+SET ROLE alter_role_1;
+SELECT run_command_on_workers('SHOW enable_hashjoin');
+ run_command_on_workers
+---------------------------------------------------------------------
+ (localhost,57637,t,off)
+ (localhost,57638,t,off)
+(2 rows)
+
+-- make sure that only alter_role_1 was affected
+RESET ROLE;
+SELECT run_command_on_workers('SHOW enable_hashjoin');
+ run_command_on_workers
+---------------------------------------------------------------------
+ (localhost,57637,t,on)
+ (localhost,57638,t,on)
+(2 rows)
+
+-- RESET ALL with IN DATABASE clause
+ALTER ROLE alter_role_1 IN DATABASE regression RESET ALL;
+ALTER ROLE alter_role_1 RESET ALL;
+ALTER ROLE ALL RESET ALL;
+-- FROM CURRENT clauses
+SET statement_timeout TO '1min';
+ALTER ROLE alter_role_1 SET statement_timeout FROM CURRENT;
+SET ROLE alter_role_1;
+SELECT run_command_on_workers('SHOW statement_timeout');
+  run_command_on_workers
+---------------------------------------------------------------------
+ (localhost,57637,t,1min)
+ (localhost,57638,t,1min)
+(2 rows)
+
+RESET statement_timeout;
+RESET ROLE;
+-- the session defaults should be updated on master_add_node
+SELECT master_remove_node('localhost', :worker_1_port);
+ master_remove_node
+---------------------------------------------------------------------
+
+(1 row)
+
+ALTER ROLE SESSION_USER SET enable_mergejoin TO false;
+ALTER ROLE CURRENT_USER SET statement_timeout TO '1.5min';
```

Comment:
Looking at CI this seems unsupported on pg11, lets make it `2min` or something that is accepted by pg11 at least